json-schema-core/src/main/java/io/ballerina/jsonschema/core/Generator.java

```diff
@@ -167,11 +230,504 @@ String convert(Object schemaObject, String name) throws Exception {
                 })
                 .collect(Collectors.joining(PIPE));
 
-        ID_TO_TYPE_MAP.put(schema.getIdKeyword(), typeName);
         return typeName;
     }
 
-    private String generateStringRepresentation(Object obj) throws InvalidDataTypeException {
+    private String createType(String name, Schema schema, Object type) throws Exception {
+        if (type == null) {
+            return NULL;
+        }
+        if (type == Boolean.class) {
+            return BOOLEAN;
+        }
+        if (type == Long.class) {
+            return createInteger(name, schema.getMinimum(), schema.getExclusiveMinimum(), schema.getMaximum(),
+                    schema.getExclusiveMaximum(), schema.getMultipleOf());
```

Comment:
For these methods, I would just let the method be called as

```suggestion
            return createInteger(name, schema);
```

and extract the necessary fields there. Can be two overloaded methods too.